spec/11-annotations.md

```diff
@@ -39,17 +39,28 @@ String @local                               // Type annotation
 
 ### Java Platform Annotations
 
+Platform annotations are found in the `scala.annotation` package,
+and also in the `scala` package.
+
 The meaning of annotation clauses is implementation-dependent. On the
 Java platform, the following annotations have a standard meaning.
 
   * `@transient` Marks a field to be non-persistent; this is
-    equivalent to the `transient`
-    modifier in Java.
+    equivalent to the `transient` modifier in Java.
 
   * `@volatile` Marks a field which can change its value
     outside the control of the program; this
-    is equivalent to the `volatile`
-    modifier in Java.
+    is equivalent to the `volatile` modifier in Java.
+
+  * `@tailrec` Marks a method which must be transformed by the compiler
+    to eliminate self-recursive invocations in tail position.
+    It is an error if there are no such invocations, or a recursive call
+    not in tail position.
+
+  * `@switch` Marks the expression submitted to a match as "switchable",
+    such that the match can be compiled to an efficient form.
+    The compiler will warn if the type of the expression is not a switchable type.
+    Certain degenerate matches may remain unoptimized without a warning.
```

Comment:
These two are not specific to the Java Platform. They should go in a separate section for platform-agnostic annotations.